[libc] added yaml_combined files, frontend for new headergen

aaryanshukla opened the pull request:
These are the yaml files that have combined functions from different libraries
Rose will be adding the rest of the yaml in the next pr
Missing assert.yaml -- work in progress

aaryanshukla committed:
[libc] added yaml_combined files, frontend hdrgen

these are the yaml files that have combined functions from different
libraries
Rose will be adding the rest of the yaml in the next pr
Missing assert.yaml -- work in progress

SchrodingerZhu commented on libc/newhdrgen/yaml_combined/string.yaml:
```diff
@@ -0,0 +1,386 @@
+
+
+
+
```
too many newlines?

SchrodingerZhu commented on libc/newhdrgen/yaml_combined/strings.yaml:
```diff
@@ -0,0 +1,72 @@
+      - type: const char *
+      - type: int
+    guard: null
+    attributes: []
```
new line

aaryanshukla committed:
addressed comments regarding linespacing

michaelrj-google commented on libc/newhdrgen/yaml_combined/ctype.yaml:
```diff
@@ -0,0 +1,135 @@
+    return_type: int
+    arguments:
+      - type: int
+    guard: null
```
in future it'd be good to make it so that if an optional member (e.g. `guard`) isn't listed it defaults to `null` or an empty list or similar to cut down on boilerplate. It's not necessary for this patch but it's good to consider for the future.

michaelrj-google reviewed (approved):
LGTM as an initial set

aaryanshukla committed:
addressed comments regarding guard/attributes

- reduced boilerplate and changed script to update
- updated function.py to handle guard/attributes as none

aaryanshukla committed:
added black formatting for python files

RoseZhang03 commented on libc/newhdrgen/yaml_combined/sys_mman.yaml:
```diff
@@ -0,0 +1,118 @@
+#known as sys/mman in POSIX 
```
delete comment

RoseZhang03 commented on libc/newhdrgen/yaml_combined/sys_mman.yaml:
```diff
@@ -0,0 +1,118 @@
+      - type: void *
+      - type: size_t
+      - type: int
+  # this is a linux function
```
delete comment

RoseZhang03 commented on libc/newhdrgen/yaml_combined/sys_wait.yaml:
```diff
@@ -0,0 +1,33 @@
+header: sys-wait.h #sys/wait.h
```
delete comment

aaryanshukla replied:
will delete once all the yaml files are finalized in case any changes occur in between.

aaryanshukla committed:
added empty list to attributes in function.py

-attributes is being appended to the result, so if empty we need to use
an empty list, else error
- isinstance used in arguments in the case arguments is []

RoseZhang03 commented on libc/newhdrgen/yaml_combined/pthread.yaml:
```diff
@@ -0,0 +1,371 @@
+      - type: void *
+    guard: null
+    attributes:
+      - noreturn
```
this should go in return_type as __Noreturn void (spec.td file says to treat noreturn as a type)

aaryanshukla replied:
fixed!

aaryanshukla committed:
changed return types in pthread.yaml noreturn
